docs/ruleset_functions/get_BPF_building_area_types_and_zones.md

```diff
@@ -0,0 +1,84 @@
+ 
+**Function Call:**
+- **get_zone_BPF_BAT**
+
+## Logic:  
+
+- create a dictionary to map lighting building area types to BPF building area types: ```building_area_lookup = {
+	AUTOMOTIVE_FACILITY: ALL_OTHER,
+	CONVENTION_CENTER: ALL_OTHER,
+	COURTHOUSE: ALL_OTHER,
+	DINING_BAR_LOUNGE_LEISURE: RESTAURANT,  
+	DINING_CAFETERIA_FAST_FOOD: RESTAURANT,
+	DINING_FAMILY: RESTAURANT,
+	DORMITORY: MULTIFAMILY,
+	EXERCISE_CENTER: ALL_OTHER,  
+	FIRE_STATION: ALL_OTHER,  
+	GYMNASIUM: ALL_OTHER,  
+	HEALTH_CARE_CLINIC: HEALTHCARE_HOSPITAL,  
+	HOSPITAL: HEALTHCARE_HOSPITAL,  
+	HOTEL_MOTEL: HOTEL_MOTEL,  
+	LIBRARY: ALL_OTHER,  
+	MANUFACTURING_FACILITY: ALL_OTHER,  
+	MOTION_PICTURE_THEATER: ALL_OTHER,  
+	MULTIFAMILY: MULTIFAMILY,
+	MUSEUM: ALL_OTHER,  
+	OFFICE: OFFICE,  
+	PARKING_GARAGE: ALL_OTHER,  
+	PENITENTIARY: ALL_OTHER,  
+	PERFORMING_ARTS_THEATER: ALL_OTHER,  
+	POLICE_STATION: ALL_OTHER,  
+	POST_OFFICE: ALL_OTHER,  
+	RELIGIOUS_FACILITY: ALL_OTHER,  
+	RETAIL: RETAIL,  
+	SCHOOL_UNIVERSITY: SCHOOL,  
+	SPORTS_ARENA: ALL_OTHER,
+	TOWN_HALL: OFFICE,  
+	TRANSPORTATION: ALL_OTHER,  
+	WAREHOUSE: WAREHOUSE,  
+	WORKSHOP: WAREHOUSE,  
+	NONE: NONE, 
+	}```
+
+- Create a dict to store results`building_area_types_with_total_area_and_zones_dict`: `building_area_types_with_total_area_and_zones_dict = {}`
+- For each building segment in RMD: `for building_segment in RMD.building.building_segments:`
+	- If the building_segment lighting_building_area_type is defined: `if building_segment.lighting_building_area_type != NULL and building_segment.lighting_building_area_type != NONE`
+		- Assign the BPF_BAT to the building_segment_BPF_BAT using the lighting_space_lookup: `building_segment_BPF_BAT = building_area_lookup[building_segment.lighting_building_area_type]`
+		- Assign "BUILDING_SEGMENT_LIGHTING" to classification_source: `classification_source = "BUILDING_SEGMENT_LIGHTING"`
+	- Else look at each zone, and then each space and determine the building_segment_BPF_BAT using the largest space.lighting_space_type: `else:`
+		- Create a dictionary to keep track of the space types and areas: `building_segment_space_types_areas_dict = {}`
+		- For each zone in the building segment: `for zone in building_segment.zones:`
+			- Create a dictionary for the zone BPF_BAT space types using the function get_zone_BPF_BAT: `zone_BPF_BAT_dict = get_zone_BPF_BAT(B_RMD, zone.id)`
+			- For each mapped building area type in the zone_BPF_BAT_dict: `for space_BPF_BAT in zone_BPF_BAT_dict:`
+				- If the BAT doesn't exist yet: Add this BAT to the building_segment_space_types_areas_dict with its area: `if space_BPF_BAT not in building_segment_space_types: building_segment_space_types_areas_dict[space_BPF_BAT] = zone_BPF_BAT_dict[space_BPF_BAT]`
```

Comment:
Should this be "if space_BPF_BAT not in building_segment_space_types**_areas_dict**" instead?